editor/components/post-taxonomies/hierarchical-term-selector.js

```diff
@@ -216,14 +216,17 @@ class HierarchicalTermSelector extends Component {
 	}
 
 	render() {
+		const { label, taxonomy, instanceId } = this.props;
+		if ( ! taxonomy.data ) {
+			return null;
+		}
 		const { availableTermsTree, availableTerms, formName, formParent, loading, showForm } = this.state;
-		const { label, slug, instanceId } = this.props;
-
-		const newTermButtonLabel = slug === 'category' ? __( 'Add new category' ) : __( 'Add new term' );
-		const newTermLabel = slug === 'category' ? __( 'Category Name' ) : __( 'Term Name' );
-		const parentSelectLabel = slug === 'category' ? __( 'Parent Category' ) : __( 'Parent Term' );
-		const noParentOption = slug === 'category' ? _x( 'None', 'category' ) : _x( 'None', 'term' );
-		const newTermSubmitLabel = slug === 'category' ? __( 'Add Category' ) : __( 'Add Term' );
+		const { labels } = taxonomy.data;
+		const newTermButtonLabel = labels.add_new_item;
```

Comment:
Would we want a fallback value to be shown in the case that the data is not loaded yet?

Reply:
Hi @aduth, fallback labels were added.